test/sscg_generator_http_SUITE.erl

```diff
@@ -0,0 +1,63 @@
+-module(sscg_generator_http_SUITE).
+
+-behaviour(ct_suite).
+-include_lib("stdlib/include/assert.hrl").
+
+% Define
+-define(MOCK_URL, <<"http://mock.endpoint">>).
+-define(MOCK_JSONDATA, #{}).
+
+% Test Setup
+-export([all/0, init_per_testcase/2, end_per_testcase/2]).
+
+% Test Cases
+-export([successful_request_test/1,
+         unexpected_status_code_test/1,
+         request_failure_test/1]).
+
+%--- Test Setup ----------------------------------------------------------------
+all() ->
+    [successful_request_test,
+     unexpected_status_code_test,
+     request_failure_test].
+
+init_per_testcase(_TestCase, Config) ->
+    meck:new(hackney, [passthrough]),
+    meck:new(jsone, [passthrough]),
+    Config.
+
+end_per_testcase(_TestCase, Config) ->
+    meck:unload(hackney),
+    meck:unload(jsone),
+    Config.
+
+%--- Test Cases ----------------------------------------------------------------
+
+successful_request_test(_Config) ->
+    meck:expect(jsone, encode, fun(_Data, _Opts) -> <<"{}">> end),
```

Comment:
Could you explain what's the point of mocking `json:encode/2`?

Reply:
I did it because I wanted to trigger a badarg error and test the error case, cause I couldn’t create a map with invalid content ([Issue 88](https://github.com/sile/jsone/issues/88)). Then I forgot to remove it afterward :(